processManager/processManager-war/src/main/java/org/silverpeas/processmanager/ProcessManagerSessionController.java

```diff
@@ -387,21 +387,21 @@ public boolean isUserAllowedOnActiveStates() {
         QualifiedUsers workingUsers = state.getWorkingUsers();
 
         RelatedUser[] relatedUsers = workingUsers.getRelatedUsers();
-        String role = "";
+        StringBuilder role = new StringBuilder();
         if (relatedUsers != null) {
           for (RelatedUser relatedUser : relatedUsers) {
             if (role.length() > 0) {
-              role += ", ";
+              role.append(", ");
             }
             // Process participants
             Participant participant = relatedUser.getParticipant();
             String relation = relatedUser.getRelation();
             if (participant != null && relation == null) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS134)